### PR TITLE
Add support for external escape valve (`^dir`)

### PR DESCRIPTION
**Pull request by wycats**
This commit makes it possible to force nu to treat a command as an external command by prefixing it with `^`. For example `^dir` will force `dir` to run an external command, even if `dir` is also a registered nu command.

This ensures that users don't need to leave nu just because we happened to use a command they need.

This commit adds a new token type for external commands, which, among other things, makes it pretty straight forward to syntax highlight external commands uniquely, and generally to treat them as special.

### CODE REVIEW

File: tests/external_tests.rs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+mod helpers;
+
+use helpers::in_directory as cwd;
+
+#[test]
+fn external_command() {
+    // Echo should exist on all currently supported platforms. A better approach might
+    // be to generate a dummy executable as part of the tests with known semantics.
+    nu!(output, cwd("tests/fixtures"), "echo 1");
```
Comment (jonathandturner) [inline tests/external_tests.rs]: Should this be ^echo?

Comment (wycats) [reply tests/external_tests.rs]: I meant to have two tests -- one `echo` and one `^echo`. Good catch.

Comment (androbtech) [reply tests/external_tests.rs]: Let's land this. I'll work on the idea listed in the comment so you guys can keep working on the other things.

Comment (jonathandturner) [reply tests/external_tests.rs]: @androbtech done

**Pull request merged**